typescript/core/src/ChunkCursor.ts

```diff
@@ -36,7 +25,8 @@ export class ChunkCursor {
   #endTime: bigint | undefined;
   #reverse: boolean;
 
-  #messageIndexCursors?: Heap<MessageIndexCursor>;
+  #orderedMessageOffset?: [logTime: bigint, offset: bigint][];
+  #messageOffsetIndex = 0;
```

Comment:
Lets add a comment on this and how it is used - my understanding of this from the implementation is that it is the next message offset index that we need to use to produce/read the next message?